src/ethereum/frontier/trie.py

```diff
@@ -199,73 +375,53 @@ def patricialize(obj: Mapping[Bytes, Node], i: Uint) -> rlp.RLP:
         Root node of `obj`.
     """
     if len(obj) == 0:
-        # note: empty storage tree has merkle root:
-        #
-        #   crypto.keccak256(RLP(b''))
-        #       ==
-        #   56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 # noqa: E501,SC100
-        #
-        # also:
-        #
-        #   crypto.keccak256(RLP(()))
-        #       ==
-        #   1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347 # noqa: E501,SC100
-        #
-        # which is the sha3Uncles hash in block header for no uncles
-
-        return b""
+        return None
 
     key = next(iter(obj))  # get first key, will reuse below
 
     # if leaf node
     if len(obj) == 1:
-        leaf = obj[key]
-        node: rlp.RLP = encode_leaf(leaf)
-        return (nibble_list_to_compact(key[i:], True), node)
+        leaf = LeafNode(key[level:], obj[key])
+        return leaf
 
     # prepare for extension node check by finding max j such that all keys in
     # obj have the same key[i:j]
-    substring = copy(key)
-    j = Uint(len(substring))
+    substring = key[level:]
+    prefix_length = len(substring)
     for key in obj:
```

Comment:
I believe this for loop clobbers `key`. Just wanted to make sure that's intentional.

Reply:
All the uses of `key` outside loops are only used to interrogate properties common to all keys, so the fact that key gets clobbered haphazardly doesn't matter.

Renamed to `arbitrary_key` for clarity.